### PR TITLE
ARMADA-440 Add doc page Continuous Integration

### PR DESCRIPTION
**Pull request by fschuch**

### CODE REVIEW

**Review by julianaklulo — APPROVED** [review]
Nice work!

File: jobbergate-docs/src/conf.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
 exclude_patterns = []
 extensions = [
     "sphinx.ext.githubpages",
+    "sphinx.ext.autosectionlabel",
```
Comment (dusktreader) [inline jobbergate-docs/src/conf.py:34]: What does this do?

File: jobbergate-docs/src/developer_guide/ci.rst [hunk 2]
```diff
@@ -2,4 +2,76 @@
  Continuous Integration
 ========================
 
-TODO: Write this in concert with Felipe
+Jobbergate uses `GitHub actions <https://github.com/omnivector-solutions/jobbergate/actions>`_ as continuous integration tools. They are described in details in this page.
```
Comment (dusktreader) [inline jobbergate-docs/src/developer_guide/ci.rst:5]:
```suggestion
Jobbergate uses `GitHub actions <https://github.com/omnivector-solutions/jobbergate/actions>`_ as continuous integration tools. They are described in detail on this page.
```

File: jobbergate-docs/src/developer_guide/ci.rst [hunk 3]
```diff
@@ -2,4 +2,76 @@
...
+
+-------------------
+ Quality Assurance
+-------------------
+
+A dedicated GitHub action is included at Jobbergate's repository 
+(`test_on_push.yaml <https://github.com/omnivector-solutions/jobbergate/blob/main/.github/workflows/test_on_push.yaml>`_)
+to run the quality assurance tools for all sub-projects at once.
+It includes unit tests, code coverage, linters, code formatters, and static code checkers,
+that are detailed in the section :any:`Quality Assurance Tools`.
```
Comment (dusktreader) [inline jobbergate-docs/src/developer_guide/ci.rst:15]:
```suggestion
They include unit tests, code coverage, linters, code formatters, and static type checkers.
Each is detailed in the section :any:`Quality Assurance Tools`.
```

File: jobbergate-docs/src/developer_guide/ci.rst [hunk 4]
```diff
@@ -2,4 +2,76 @@
  Continuous Integration
 ========================
 
-TODO: Write this in concert with Felipe
+Jobbergate uses `GitHub actions <https://github.com/omnivector-solutions/jobbergate/actions>`_ as continuous integration tools. They are described in details in this page.
+
+-------------------
+ Quality Assurance
+-------------------
```
Comment (dusktreader) [inline jobbergate-docs/src/developer_guide/ci.rst:9]: For other docs pages we aren't using the top line and extra spacing for sections inside the body.

```suggestion
Quality Assurance
-----------------
```

File: jobbergate-docs/src/developer_guide/ci.rst [hunk 5]
```diff
@@ -2,4 +2,76 @@
...
+^^^^^^^^^^^^^^^^^^^^^
+ Prepare for release
+^^^^^^^^^^^^^^^^^^^^^
+
+This action
+(`prepare_release.yaml <https://github.com/omnivector-solutions/jobbergate/blob/main/.github/workflows/prepare_release.yaml>`_)
+is triggered on a workflow dispatch event and demands some interactions
+to ensure code quality. It takes as arguments:
+
+* The base branch where the action will run from. Notice that the use of the ``main``
```
Comment (dusktreader) [inline jobbergate-docs/src/developer_guide/ci.rst:40]:
```suggestion
* The base branch from which the action will run. Notice that the use of the ``main``
```

File: jobbergate-docs/src/developer_guide/ci.rst [hunk 6]
```diff
@@ -2,4 +2,76 @@
...
+  branch is highly recommended in order to keep a linear commit history between releases and pre-releases.
+* The bump version rule to be applied, based on
+  `Poetry version <https://python-poetry.org/docs/cli/#version>`_.
+  The following options are presented in a drop-down menu:
+  ``patch``, ``minor``, ``major``, ``prepatch``, ``preminor``, ``premajor``, ``prerelease``.
+
+Once activated, this action:
+
+* Uses Poetry to bump the version number of all the Jobbergate packages;
```
Comment (dusktreader) [inline jobbergate-docs/src/developer_guide/ci.rst:49]: In English, semicolons are only used in sentences to separate two independent clauses. It's not common usage to put them on the end of list items.

```suggestion
* Uses Poetry to bump the version number of all the Jobbergate packages
```

File: jobbergate-docs/src/developer_guide/ci.rst [hunk 7]
```diff
@@ -2,4 +2,76 @@
...
+* Checks if the new version number is synchronized between them, and fails if they are not;
+* Creates a new entry on each changelog file with the new version number and the current date,
+  including all unreleased features and bug fixes.
+* Creates a new branch named ``release/<version>``.
+* Opens a draft pull request titled ``Release <version>``.
+  In this way, all the changes above can be reviewed before the release is published,
+  and all quality assurance tests are executed for the pull request.
```
Comment (dusktreader) [inline jobbergate-docs/src/developer_guide/ci.rst:56]:
```suggestion

In this way, all the changes above can be reviewed before the release is published,
and all quality assurance tests are executed for the pull request.
```

**Review by dusktreader — CHANGES_REQUESTED** [review]
Nice work! Some comments for you to review.

File: jobbergate-docs/src/conf.py (hunk 1 above)
Comment (fschuch) [reply jobbergate-docs/src/conf.py:34]: This is for cross-references, like this one:

https://github.com/omnivector-solutions/jobbergate/blob/df3660a4e0aa21cc7ade37a6f3ec652501b1b1ea/jobbergate-docs/src/developer_guide/ci.rst#L15

File: jobbergate-docs/src/developer_guide/ci.rst (hunk 2 above)
Comment (fschuch) [reply jobbergate-docs/src/developer_guide/ci.rst]: solved in [58f2f91](https://github.com/omnivector-solutions/jobbergate/pull/144/commits/58f2f918da90e960892e55f68c4e15734043d298).

File: jobbergate-docs/src/developer_guide/ci.rst (hunk 3 above)
Comment (fschuch) [reply jobbergate-docs/src/developer_guide/ci.rst]: solved in [58f2f91](https://github.com/omnivector-solutions/jobbergate/pull/144/commits/58f2f918da90e960892e55f68c4e15734043d298).

File: jobbergate-docs/src/developer_guide/ci.rst (hunk 4 above)
Comment (fschuch) [reply jobbergate-docs/src/developer_guide/ci.rst]: solved in [58f2f91](https://github.com/omnivector-solutions/jobbergate/pull/144/commits/58f2f918da90e960892e55f68c4e15734043d298).

File: jobbergate-docs/src/developer_guide/ci.rst (hunk 5 above)
Comment (fschuch) [reply jobbergate-docs/src/developer_guide/ci.rst]: solved in [58f2f91](https://github.com/omnivector-solutions/jobbergate/pull/144/commits/58f2f918da90e960892e55f68c4e15734043d298).

File: jobbergate-docs/src/developer_guide/ci.rst (hunk 6 above)
Comment (fschuch) [reply jobbergate-docs/src/developer_guide/ci.rst]: solved in [58f2f91](https://github.com/omnivector-solutions/jobbergate/pull/144/commits/58f2f918da90e960892e55f68c4e15734043d298).

File: jobbergate-docs/src/developer_guide/ci.rst (hunk 7 above)
Comment (fschuch) [reply jobbergate-docs/src/developer_guide/ci.rst]: solved in [93f489e](https://github.com/omnivector-solutions/jobbergate/pull/144/commits/93f489ed7b02386e175af69af7b7f4342d7b84ec).

**Review by dusktreader — APPROVED** [review]

**Pull request merged**